typemoq/typemoq.d.ts

```diff
@@ -0,0 +1,653 @@
+// Type definitions for typemoq v0.0.6
+// Project: https://github.com/florinn/typemoq
+// Definitions by: Florin Nitoi <https://github.com/florinn>
+// Definitions: https://github.com/borisyankov/DefinitelyTyped
+
+
+declare module TypeMoq {
+    class Cons {
+        static IMATCH_ID_VALUE: string;
+        static IMATCH_ID_NAME: string;
+        static IMATCH_MATCHES_NAME: string;
+    }
+}
+
+
+declare module TypeMoq {
+    class CurrentInterceptContext<T> {
+        call: proxy.IProxyCall<T>;
+    }
+}
+
+
+declare module TypeMoq {
+    enum GlobalType {
+        Class = 0,
+        Function = 1,
+        Value = 2,
+    }
+    class GlobalMock<T> implements IGlobalMock<T> {
+        mock: Mock<T>;
+        private _name;
+        private _type;
+        container: Object;
+        constructor(mock: Mock<T>, _name: string, _type: GlobalType, container: Object);
+        static ofInstance<U>(instance: U, name?: string, container?: Object, behavior?: MockBehavior): GlobalMock<U>;
+        static ofType<U>(ctor: Ctor<U>, container?: Object, behavior?: MockBehavior): GlobalMock<U>;
+        object: T;
+        name: string;
+        behavior: MockBehavior;
+        callBase: boolean;
+        type: GlobalType;
+        setup<TResult>(expression: IFunc2<T, TResult>): MethodCallReturn<T, TResult>;
+        verify<TResult>(expression: IFunc2<T, TResult>, times: Times): void;
+        verifyAll(): void;
+    }
+}
+
+
+declare module TypeMoq.Api {
+    interface ICallback<T, TResult> {
```

Comment:
http://www.typescriptlang.org/Handbook#writing-dts-files

> Naming Conventions
In general, do not prefix interfaces with I (e.g. IColor). Because the concept of an interface in TypeScript is much more broad than in C# or Java, the IFoo naming convention is not broadly useful.